packages/core/src/internal/decorators/global-style.ts

```diff
@@ -26,18 +26,24 @@ function standardEvent(descriptor: PropertyDescriptor, element: { key: string })
  * This is useful for when styles need to be applied that are not supported
  * withing a `::slotted()` selector such as ::-vendor style selectors.
  */
-export function globalStyle(options = { static: true }) {
+export function globalStyle() {
   return (protoOrDescriptor: any, name: string): any => {
-    const setter = function (this: any, value: CSSResult | string) {
-      if (!this[`__${name}StyleTag`]) {
-        this[`__${name}StyleTag`] = document.createElement('style');
-        this.prepend(this[`__${name}StyleTag`]);
-        this[`__${name}StyleTag`].innerText = value.toString();
-      }
+    const targetConnectedCallback: () => void = protoOrDescriptor.connectedCallback;
+
+    function connectedCallback(this: any): void {
+      this[`__${name}StyleTag`] = document.createElement('style');
+      this.prepend(this[`__${name}StyleTag`]);
+      this[`__${name}StyleTag`].innerText = this[`__${name}GlobalStyle`].toString();
 
-      if (!options.static) {
-        this[`__${name}StyleTag`].innerText = value.toString();
+      if (targetConnectedCallback) {
+        targetConnectedCallback.apply(this);
       }
+    }
+
+    protoOrDescriptor.connectedCallback = connectedCallback;
+
+    const setter = function (this: any, value: CSSResult | string) {
+      this[`__${name}GlobalStyle`] = value;
```

Comment:
Instead of writing the style tag within the setter/constructor we hook into the `connectedCallback` lifecycle. This is important as the component should not set attrs or DOM on itself before `connectedCallback`. It violates the spec and some browsers with throw exceptions depending on how the element is created.